internal/host/validator.go

```diff
@@ -395,15 +395,24 @@ func (v *validator) isValidPlatform(c *validationContext) ValidationStatus {
 	if c.inventory.SystemVendor == nil {
 		return ValidationError
 	}
-	return boolValue(!funk.ContainsString(invalidPlatforms, c.inventory.SystemVendor.ProductName))
+	if funk.ContainsString(invalidPlatforms, c.inventory.SystemVendor.ProductName) {
+		// In case there is no cluster validation is pending
```

Comment:
Or late binding, @yevgeny-shnaidman @ravidbro how does refresh status will filter out validations for the pool?

Reply:
no cluster == late binding no?

Reply:
currently validations are either containing cluster (day1, day2) or infra-env (pool). To filter out validation for pool, simply check if infraenv in validation context s not null. see example:
https://github.com/openshift/assisted-service/blob/8f89ccc17c91559b00d9b5e49d4dacc1d63ad526/internal/host/validator.go#L338

Reply:
`no cluster` sounds clearer then `infraenv == nil`, i would like to make sure that we don't have two different ways in the code.

Reply:
currently all validations are using infraenv == nil. You can change it as you like , as long as it is the same for all validaitons

Reply:
I'm not sure what c.infraEnv != nil means (that's what we check in isMachineCidrDefined !)
```
func (v *validator) isMachineCidrDefined(c *validationContext) ValidationStatus {
	if c.infraEnv != nil {
		return ValidationSuccessSuppressOutput
	}
```
If c.infraEnv == nil host beolngs to a cluster?

In order to check `c.cluster.UserManagedNetworking` I need to make sure `c.cluster != inl` anyway.
So I prefer to keep the code as is (assuming logic is correct)

Reply:
What does this mean?  How can a host not have an InfraEnv?  @ravidbro any idea?

Reply:
yes, because of a logic in https://github.com/openshift/assisted-service/blob/master/internal/host/validator.go#L171-L227

Reply:
Suggesting to use `if c.infraEnv != nil` here and then fix everything with https://issues.redhat.com/browse/MGMT-7459